Modelica/Math/isPowerOf2.mo

```diff
@@ -5,7 +5,7 @@ function isPowerOf2 "Determine if the integer input is a power of 2"
   output Boolean result "= true, if integer scalar is a power of 2";
 algorithm
   assert(i>=1, "Integer input to isPowerOf2 has to be >= 1");
-  result := i == 2^integer(log(i)/log(2)+0.5);
+  result := i == integer(2^integer(log(i)/log(2)+0.5)+0.5);
```

Comment:
As described in https://github.com/modelica/ModelicaStandardLibrary/issues/3836#issuecomment-858652806, I don't think the added `+ 0.5` is needed.  The fact that it isn't needed is related to the idea behind this working at all for the large values `i` that this is all about, so the `+ 0.5` could actually give a misleading idea of what is really going on.

A better idea could be to explain the trick in a comment:
```suggestion
  /* Note that 2 ^ <integer-value> will be represented exactly as a floating point number
   * even for large exponents.  When converted to Integer using the 'integer' function, this
   * gives the exact Integer power of 2 even though the precision in the intermediate floating
   * point representation might seem insufficient.
   */
  result := i == integer(2 ^ integer(log(i) / log(2) + 0.5));
```

Reply:
But `2^n` isn't strictly defined in Modelica to be accurate for integers, and the extra 0.5 guards against that. IEEE 754-2008 states that pow should give correctly rounded results, but Modelica doesn't require that, and there have been bad C-implementations with slightly inaccurate pow-functions.

And it still doesn't say what the original error would be. 

The largest 32-bit signed integer is 2147483647 =2^31-1 and Modelica.Math.isPowerOf2(2147483647) returns false, as the rhs is 2^31=2147483648.0 which is a Real number and not equal to the lhs 2147483647.

If we use 64-bit integers, and 64-bit Reals the entire algorithm needs to be rewritten and just adding an integer is meaningless, since the integer value `2^60` simply cannot be represented as a 64-bit Real.

Reply:
> IEEE 754-2008 states that pow should give correctly rounded results, but Modelica doesn't require that, and there have been bad C-implementations with slightly inaccurate pow-functions.

But doesn't that just mean that the Modelica definition of `pow` needs to be improved?  I mean, if one can't assume correctly rounded results, isn't it just ridiculous to have it in the core language then?  If one wants to call the `pow` in the C standard library, nothing more, nothing less, then it would make more sense to just wrap it in an external function – and this use-at-your-own-risk function should probably not even be exposed through the MSL.

OK, I made an invalid assumption about `pow` giving correctly rounded results.  But can you at least see that a correctly rounding `pow` is actually sufficient to also handle very big integer representations?  I mean, take a `double` with 11 bit exponent.  It can represent powers of 2 from 2^0 to 2^1023 exactly, so you'd need a very wide integer type before the algorithm would break down.

So, the error that this fixes – at least on my setup with a sound `pow` function – is that somewhere around 2^53, I'm starting to get `true` also for numbers that are just close to a power of 2, since the conversion of both sides of the `==` to `Real` is throwing away bits that I need to get correct results.